### PR TITLE
Release notes for clustering API

### PR DESCRIPTION
**Pull request by benkli01**
Adding release notes for the new clustering API

### CODE REVIEW

**Comment by alanchiao** [comment]
Actually could you move clustering to the top of the release notes and then squash the commits? 

Given that this is clustering's first release, it makes sense to highlight it at the top.

File: RELEASE.md [hunk 1]
```diff
@@ -23,4 +23,11 @@ Keras pruning API:
 
  * Tested against TensorFlow 1.14.0, 2.0.0, and nightly.
 
-
+Keras clustering API:
+
+* New API for clustering of weights
+* Major Features
+  * Support for clustering of convolutional (except DepthwiseConv), fully connected and other commonly used standard Keras layers
```
Comment (alanchiao) [inline RELEASE.md:30]: fully connected -> Dense to match Keras terminology.

File: RELEASE.md [hunk 2]
```diff
@@ -23,4 +23,11 @@ Keras pruning API:
 
  * Tested against TensorFlow 1.14.0, 2.0.0, and nightly.
 
-
+Keras clustering API:
+
+* New API for clustering of weights
```
Comment (arovir01) [inline RELEASE.md:28]: clustering of weights -> weight clustering

File: RELEASE.md (hunk 1 above)
Comment (arovir01) [inline RELEASE.md:30]: clustering of convolutional <- the "of" is not needed here

Comment (benkli01) [reply RELEASE.md:30]: Ok, I will change that.

File: RELEASE.md (hunk 2 above)
Comment (benkli01) [reply RELEASE.md:28]: I think both work fine, but _clustering of weights_ seems to be more precise to me, but it's rather subjective, isn't it?

File: RELEASE.md (hunk 1 above)
Comment (benkli01) [reply RELEASE.md:30]: I will update this and also move the whole block to the top. Thanks for the feedback.

File: RELEASE.md (hunk 2 above)
Comment (benkli01) [reply RELEASE.md:28]: I will change it to _weight clustering_ to be more consistent with the documentation.

**Comment by alanchiao** [comment]
Merging now. Release notes can be modified, even after the release is done, if desired.

**Pull request merged**